src/App.vue

```diff
@@ -104,23 +117,9 @@ export default {
 		 */
 		EventBus.$once('conversationsReceived', () => {
 			if (this.$route.name === 'conversation') {
-				const CURRENT_CONVERSATION_NAME = this.getConversationName(this.token)
-				this.setPageTitle(CURRENT_CONVERSATION_NAME)
+				this.setPageTitle(this.currentConversationName)
 			}
 		})
-		/**
-		 * Global before guard, this is called whenever a navigation is triggered.
-		*/
-		Router.beforeEach((to, from, next) => {
-			/**
-			 * This runs whenever the new route is a conversation.
-			 */
-			if (to.name === 'conversation') {
-				const NEXT_CONVERSATION_NAME = this.getConversationName(to.params.token)
-				this.setPageTitle(NEXT_CONVERSATION_NAME)
-			}
-			next()
-		})
```

Comment:
I think that we should keep the Router.beforeEach approach. 
Watchers are expensive and in the future we might need to trigger other things within `Router.beforeEach` ( even outside of `conversations` named routes ).